src/mechanics/mechanics-blocks.ts

```diff
@@ -306,22 +307,31 @@ export class MechanicsRenderer {
 
   async renderProgress(target: HTMLElement, node: KdlNode) {
     const trackName = (node.properties.name ?? node.values[0]) as string;
-    let from = node.properties.from as number;
-    const fromBoxes =
-      (node.properties["from-boxes"] as number) ??
-      (from != null ? Math.floor(from / 4) : 0);
-    const fromTicks =
-      (node.properties["from-ticks"] as number) ??
-      (from != null ? from % 4 : 0);
-    if (from == null) {
-      from = fromBoxes * 4 + fromTicks;
+    const result = ProgressTrack.create({
+      progress:
+        node.properties.from ??
+        ((node.properties["from-boxes"] as number) ?? 0) * 4 +
+          ((node.properties["from-ticks"] as number) ?? 0),
+      difficulty: node.properties.level,
```

Comment:
Is was meaning to change this: the correct name for this, I believe, is "rank", so maybe we can make that change right here, too. We'll want to fall back to "level" for backwards compat, though.

Reply:
Ah, yeah, you are right. I think `Difficulty` came from the original Forged in Obsidian track format, and I kept it for backwards compatibility, but I, too, had intended to change the name. I'll make that change in this PR in both spots.